src/domain/usecases/TrackMatomoViewUseCase.ts

```diff
@@ -0,0 +1,23 @@
+import { MatomoAnalyticsRepository } from "../../data/repositories/MatomoAnalyticsRepository";
+import { AnalyticsEvent } from "../entities/AnalyticsEvent";
+import { AnalyticsConfigRepository } from "../repositories/AnalyticsConfigRepository";
+
+export class TrackMatomoViewUseCase {
+    constructor(
+        private configRepository: AnalyticsConfigRepository,
+        private matomoAnalytics: MatomoAnalyticsRepository
+    ) {}
+
+    public async execute(options?: { location?: string; title?: string }) {
```

Comment:
We have both location and title as optional, which also has it's own handling logic whenever they're not defined (on the use case). Something that on other apps I see it as a presentation thing as each component could send an event with specific (or not) location / title (for example in UHCPW), here on HomePage app we're defining location and title each time we track a page view (and events in the future) as it is a generic app. Do you think we should search for location and title optional args along the code and remove the use optional?